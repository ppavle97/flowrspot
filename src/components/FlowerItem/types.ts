export interface StyledFlowerItemProps {
  favourite?: boolean;
  profile_picture: string;
}

export interface FlowerItemProps {
  favourite?: boolean;
  name: string;
  latin_name: string;
  sightings: string;
  profile_picture: string;
  onFavouriteClick: () => void;
}
