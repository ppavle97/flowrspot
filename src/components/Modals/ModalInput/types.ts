import React from "react";

export interface ModalInputProps {
  state: any;
  setState: any;
  label: string;
  type: string;
}
