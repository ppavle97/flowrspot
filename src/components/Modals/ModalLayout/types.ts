import { ReactNode } from "react";

export interface ModalLayoutProps {
  children: ReactNode;
}
