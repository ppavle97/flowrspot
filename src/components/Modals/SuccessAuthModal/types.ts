export interface SuccessAuthModalProps {
  description: string;
  type: string;
}
