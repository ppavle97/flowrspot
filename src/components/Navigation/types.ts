export interface StyledMenuButtonProps {
  login?: boolean;
}

export interface StyledMenuProps {
  menuOpen?: boolean;
}
export interface StyledNavigationProps {
  menuOpen?: boolean;
}
