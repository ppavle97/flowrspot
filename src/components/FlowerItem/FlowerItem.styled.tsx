import styled from "@emotion/styled";
import { StyledFlowerItemProps } from "./types";
import { breakpointMax, calcRem, colors } from "../../theme";

export const StyledFlowerItem = styled("div")<StyledFlowerItemProps>(
  ({ favourite, profile_picture }) => [
    {
      backgroundImage: `url(${profile_picture})`,
      backgroundSize: "cover",
      backgroundPosition: "bottom",
      textAlign: "center",
      boxSizing: "border-box",
      height: calcRem(400),
      position: "relative",
      display: "flex",
      alignItems: "flex-end",
      justifyContent: "center",

      [breakpointMax("md")]: {
        height: calcRem(300),
      },
      [breakpointMax("sm")]: {
        height: calcRem(230),
      },

      "& .favourite": {
        width: calcRem(40),
        height: calcRem(40),
        backgroundColor: favourite ? colors.lightCoral : colors.white,
        borderRadius: "100px",
        position: "absolute",
        top: calcRem(20),
        right: calcRem(20),
        display: "flex",
        alignItems: "center",
        justifyContent: "center",
        "&:hover": { cursor: "pointer" },

        [breakpointMax("md")]: {
          width: calcRem(30),
          height: calcRem(30),
          top: calcRem(15),
          right: calcRem(15),
        },
        [breakpointMax("sm")]: {
          width: calcRem(25),
          height: calcRem(25),
          top: calcRem(10),
          right: calcRem(10),
        },

        "& img": {
          width: calcRem(20),
          [breakpointMax("md")]: {
            width: calcRem(15),
          },
          [breakpointMax("sm")]: {
            width: calcRem(12),
          },
        },
      },
      "& .content": {
        paddingBottom: calcRem(10),

        "& h3": {
          color: colors.white,
          fontSize: calcRem(20),
          margin: 0,
          [breakpointMax("sm")]: {
            fontSize: calcRem(16),
          },
        },

        "& .latin": {
          color: colors.white,
          fontSize: calcRem(12),
          fontStyle: "italic",
          marginTop: calcRem(5),
          opacity: 0.7,
          [breakpointMax("sm")]: {
            fontSize: calcRem(10),
          },
        },

        "& .sightings": {
          color: colors.white,
          fontSize: calcRem(12),
          backgroundColor: favourite ? colors.lightCoral : "rgba(0,0,0,.3)",
          width: calcRem(110),
          margin: `${calcRem(20)} auto 0`,
          height: calcRem(35),
          display: "flex",
          alignItems: "center",
          justifyContent: "center",
          borderRadius: calcRem(30),
          [breakpointMax("sm")]: {
            fontSize: calcRem(10),
          },
        },
      },
    },
  ]
);
