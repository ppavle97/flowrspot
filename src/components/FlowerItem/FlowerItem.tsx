import { StyledFlowerItem } from "./FlowerItem.styled";
import { FlowerItemProps } from "./types";

import whiteStar from "./assets/whiteStar.svg";
import silverStar from "./assets/silverStar.svg";
import { useSelector } from "react-redux";
import { RootState } from "../../store";

export const FlowerItem = (props: FlowerItemProps) => {
  const {
    favourite,
    name,
    latin_name,
    sightings,
    profile_picture,
    onFavouriteClick,
  } = props;

  const isLoggedIn = useSelector((state: RootState) => state.user.isLoggedIn);

  return (
    <StyledFlowerItem profile_picture={profile_picture} favourite={favourite}>
      {isLoggedIn && (
        <div className="favourite" onClick={onFavouriteClick}>
          <img
            src={favourite ? whiteStar : silverStar}
            alt="FlowrSpot Favourite flower"
          />
        </div>
      )}
      <div className="content">
        <h3>{name}</h3>
        <p className="latin">{latin_name}</p>
        <p className="sightings">{sightings}</p>
      </div>
    </StyledFlowerItem>
  );
};
