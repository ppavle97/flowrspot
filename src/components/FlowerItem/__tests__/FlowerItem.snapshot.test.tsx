import { FlowerItemProps } from "../types";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { FlowerItem } from "../FlowerItem";
import { render } from "@testing-library/react";
import { screen } from "@testing-library/react";

const mockStore = configureStore([]);
describe("FlowerItem component snapshot test", () => {
  const defaultProps: FlowerItemProps = {
    favourite: false,
    name: "Rose",
    latin_name: "Latin Name",
    sightings: "10",
    profile_picture: "./assets/flower.svg",
    onFavouriteClick: jest.fn(),
  };

  const notLoggedInStore = mockStore({
    user: { isLoggedIn: false },
  });

  const store = mockStore({
    user: { isLoggedIn: true },
  });
  
  test("matches snapshot", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <FlowerItem {...defaultProps} />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("renders favourite button when user is logged in", () => {
    render(
      <Provider store={store}>
        <FlowerItem {...defaultProps} />
      </Provider>
    );
    const favouriteButton = screen.getByTestId("favourite");
    expect(favouriteButton).toBeInTheDocument();
  });

  test("does not render favourite button when user is not logged in", () => {
    render(
      <Provider store={notLoggedInStore}>
        <FlowerItem {...defaultProps} />
      </Provider>
    );
    const favouriteButton = screen.queryByTestId("favourite");
    expect(favouriteButton).not.toBeInTheDocument();
  });
});
