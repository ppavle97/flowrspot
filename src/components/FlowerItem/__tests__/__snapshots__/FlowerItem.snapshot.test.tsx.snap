// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlowerItem component snapshot test matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1nnb2kx"
  >
    <div
      class="favourite"
      data-testid="favourite"
    >
      <img
        alt="FlowrSpot Favourite flower"
        src="silverStar.svg"
      />
    </div>
    <div
      class="content"
    >
      <h3>
        Rose
      </h3>
      <p
        class="latin"
      >
        Latin Name
      </p>
      <p
        class="sightings"
      >
        10
      </p>
    </div>
  </div>
</DocumentFragment>
`;
